syntax = "proto3";

package tensorflow;
option cc_enable_arenas = true;

import "tensorflow/core/framework/types.proto";
import "tensorflow/core/framework/tensor_shape.proto";

message CppShapeInferenceResult {
  TensorShapeProto shape = 1;
  TensorShapeProto handle_shape = 2;
  DataType handle_dtype = 3;
}

message CppShapeInferenceInputsNeeded {
  repeated int32 input_tensors_needed = 1;
  repeated int32 input_tensors_as_shapes_needed = 2;
}
